plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/glue/TestHiveGlueMetastore.java

```diff
@@ -1332,6 +1332,40 @@ public void testTableWithoutStorageDescriptor()
         }
     }
 
+    /**
+     * This test requires that the permission policy on the AWS Glue to be manually modified to
+     * explicitly deny the `glue:GetTable` actions on the `gettables-permission-denied-test-database`
+     * Glue Database.
+     * </p>
+     * The manual customization of the Glue permission policies can be done via:
+     * <ul>
+     *     <li><code>aws glue put-resource-policy</code> AWS CLI operation</li>
+     *     <li><i>Catalog Settings</i> form via AWS Glue Web Console</li>
+     * </ul>
+     *
+     * The template for the permission policy is presented below:
+     * <pre>
+     * <code>
+     *     {
+     *       "Version" : "2008-10-17",
+     *       "Statement" : [ {
+     *         "Sid" : "GetTablesAccessDeny",
+     *         "Effect" : "Deny",
+     *         "Principal" : "*",
+     *         "Action" : "glue:GetTable",
+     *         "Resource" : "arn:aws:glue:AWS_REGION:AWS_ACCOUNT_ID:database/gettables-permission-denied-test-database"
+     *       }]
+     *     }
+     * </code>
+     * </pre>
+     */
+    @Test
+    public void testGetAllEntitiesOnAccessDeniedDatabase()
```

Comment:
@pettyjamesm do you have any insights about how this fix can be cleanly tested without using a manually defined Glue permission policy?

While trying to programmatically change the AWS Glue permission policies we've (see https://github.com/trinodb/trino/pull/14849 ) bumped into the constraint that the changes perform on the Glue Data Catalog programmatically may overwrite either configuration AWS Glue configuration affecting builds which run on the same time or remove existing AWS Glue permission policies which were previously manually configured.

Reply:
I think your options are basically either:
- mock the AWS client to throw access denied in the specific test
- manually define a separate Glue permission policy / IAM role for this purpose in test
- use a "create if not exists" pattern to programatically define an IAM role / Glue table with the specific required permissions such that repeated / concurrent runs will only create it once and therefore shouldn't generally interfere with each other while also allowing new environments to create the required objects on-demand during the first run